prow/cmd/plank/main.go

```diff
@@ -141,17 +155,54 @@ func main() {
 	if reporter != nil {
 		interrupts.Run(reporter)
 	}
+	if err := mgr.Start(interrupts.Context().Done()); err != nil {
+		logrus.WithError(err).Fatal("failed to start manager")
+	}
+}
+
+func v1Main(o options, mgr manager.Manager, cfg config.Getter) error {
+
+	buildClusterClients, err := o.kubernetes.BuildClusterUncachedRuntimeClients(o.dryRun)
+	if err != nil {
+		return fmt.Errorf("failed to consturct build cluster clients: %w", err)
+	}
+	c, err := plank.NewController(mgr.GetClient(), buildClusterClients, nil, cfg, o.totURL, o.selector)
+	if err != nil {
+		return fmt.Errorf("failed to create plank controller: %w", err)
+	}
+
 	interrupts.TickLiteral(func() {
 		start := time.Now()
 		c.SyncMetrics()
 		logrus.WithField("metrics-duration", fmt.Sprintf("%v", time.Since(start))).Debug("Metrics synced")
 	}, 30*time.Second)
-
 	// run the controller
 	if err := mgr.Add(c); err != nil {
-		logrus.WithError(err).Fatal("failed to add controller to manager")
+		return fmt.Errorf("failed to add controller to manager: %w", err)
 	}
-	if err := mgr.Start(interrupts.Context().Done()); err != nil {
-		logrus.WithError(err).Fatal("failed to start manager")
+
+	return nil
+}
+
+func v2Main(o options, mgr manager.Manager, cfg config.Getter) error {
```

Comment:
Why should we prefer a `plank` binary with a `--v2` flag over deploying a controller manager and enabling this controller?

Reply:
So your suggestion here is to make this the starting point of a new `prow-controller-manager` and not touch `cmd/plank` at all?

Reply:
That seems like it would create the same amount of operational change when someone wanted to opt in, and would set us up for the future.

Reply:
Done